### PR TITLE
Add a draft Roadmap.

### PR DESCRIPTION
**Pull request by jonathanpallant**
Outlines some "default" routes through the Rust Fundamentals and Applied Rust modules.

### CODE REVIEW

File: training-slides/src/roadmaps.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+### Day 1, Session 2: Defining and Implementing Traits
+
+* Implementing Traits
+* Exercise: TBD
+
+### Day 1, Session 3: Generics
+
+* Generics
+* Exercise: TBD
```
Comment (Mirabellensaft) [inline training-slides/src/roadmaps.md:71]:
```suggestion
* Exercise: Shapes, Part II
```

File: training-slides/src/roadmaps.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+* Exercise: Simple DB
+
+## Module: Applied Rust
+
+### Day 1, Session 1: Using Traits
+
+* Using Traits
+* Exercise: impl Error in SimpleDB
+
+### Day 1, Session 2: Defining and Implementing Traits
```
Comment (Mirabellensaft) [inline training-slides/src/roadmaps.md:63]: methods need to go in this too.

**Review by Mirabellensaft — APPROVED** [review]
Lifetimes are missing.

File: training-slides/src/roadmaps.md (hunk 2 above)
Comment (jonathanpallant) [inline training-slides/src/roadmaps.md:63]:
```suggestion
### Day 1, Session 2: Defining Traits and Methods
```

File: training-slides/src/roadmaps.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+## Module: Applied Rust
+
+### Day 1, Session 1: Using Traits
+
+* Using Traits
+* Exercise: impl Error in SimpleDB
+
+### Day 1, Session 2: Defining Traits and Methods
+
+* Implementing Traits
```
Comment (jonathanpallant) [inline training-slides/src/roadmaps.md:65]:
```suggestion
* Implementing Traits
* Method Syntax
```

File: training-slides/src/roadmaps.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+* Method Syntax
+* Exercise: Shapes, Part I
+
+### Day 1, Session 3: Generics
+
+* Generics
```
Comment (jonathanpallant) [inline training-slides/src/roadmaps.md:71]:
```suggestion
* Generic over Types - `<T>`
* Generic over Lifetimes - `<'a>`
```

**Comment by jonathanpallant** [comment]
> Lifetimes are missing.

Added it to "Generics" because you are generic over a lifetime like you are generic over a type.

File: training-slides/src/roadmaps.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+* Exercise: TBD
+
+### Day 3, Session 1: Closures
+
+* Closures
+* Exercise: TBD
+
+### Day 3, Session 2: Multi-threading
+
+* Spawing Threads
```
Comment (miguelraz) [inline training-slides/src/roadmaps.md:97]:
```suggestion
* Spawning Threads
```

**Review by miguelraz — APPROVED** [review]

**Pull request merged**